locator/ec2_multi_region_snitch.cc

```diff
@@ -51,12 +57,22 @@ future<> ec2_multi_region_snitch::start() {
     return seastar::async([this] {
         ec2_snitch::load_config().get();
         if (this_shard_id() == io_cpu_id()) {
-            sstring pub_addr;
             inet_address local_public_address;
 
             try {
-                pub_addr = aws_api_call(AWS_QUERY_SERVER_ADDR, AWS_QUERY_SERVER_PORT, PUBLIC_IP_QUERY_REQ).get0();
-                local_public_address = inet_address(pub_addr);
+                auto broadcast = utils::fb_utilities::get_broadcast_address();
+                if (broadcast.addr().is_ipv6()) {
+                    auto macs = aws_api_call(AWS_QUERY_SERVER_ADDR, AWS_QUERY_SERVER_PORT, PRIVATE_MAC_QUERY).get0();
+                    // CMH. should we maybe check iff there is more than one mac address? If so, should we try to find the one
+                    // associated with the public ipv4 address? What if there is none?
+                    // For now, just assume one mac -> one ipv6. 
```

Comment:
Let's check that assumption and print a warning, or fail. Otherwise we have a silent failure here that will be hard to debug.

(maybe the call below will fail due to bad format, but still it will be hard to understand)

Reply:
Ok, updated to check and warn + use first mac returned (if more than one - this is still hypothetical)